ptrace/debugger/child.py

```diff
@@ -87,21 +88,28 @@ def _createParent(pid, errpipe_read):
         raise child_exception
 
 
-def _createChild(arguments, no_stdout, env, errpipe_write):
+def _createChild(arguments,
+                 no_stdout,
+                 env,
+                 errpipe_write,
+                 close_fds=True,
+                 pass_fds=()):
     # Child code
     try:
         ptrace_traceme()
     except PtraceError as err:
         raise ChildError(str(err))
 
-    # Close all files except 0, 1, 2 and errpipe_write
-    for fd in range(3, MAXFD):
-        if fd == errpipe_write:
-            continue
-        try:
-            close(fd)
-        except OSError:
-            pass
+    if close_fds:
+        # Close all files except 0, 1, 2 and errpipe_write
+        for fd in range(3, MAXFD):
+            if fd == errpipe_write or (fd in pass_fds):
+                continue
+            try:
+                close(fd)
+            except OSError:
+                pass
```

Comment:
pass_fds file descriptors must be made inheritable with os.set_inheritable(). By default, Python creates non-inheritable FDs.

Reply:
I am confused to see `_posixsubprocess` does that, as I manually set that flag on my python-opened file descriptors.
Pushed a new commit for that.